src/insidersUpdate/gitHubInterfaces.ts

```diff
@@ -0,0 +1,93 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ */
+
+/**
+ * For reference, you can find the GitHub API information here:
+ * https://developer.github.com/v3/repos/
+ * Yes, there are "node packages" from just about everyone but they are
+ * way overkill for what we need. These interfaces are the bare-minimum
+ * needed for us to perform our upgrade checks and installs.
+ * This implementation was adapted from the Microsoft C/C++ extension.
+ * https://github.com/microsoft/vscode-cpptools/blob/58c50dc38b1a3ebcae8139b28c0904d468d11e6e/Extension/src/githubAPI.ts
+ */
+
+export const GitHubApiBase: string = 'https://api.github.com';
+
+ /**
+  * Contains the minimum information needed to inspect a git-hub release.
+  */
+export interface GitHubRelease {
+    /**
+     * The tag name for the release in the form of v<semver>-insiders.
+     * I.e. v3.2020.424006-insiders
+     */
+    readonly tag_name: string;
+
+    /**
+     * The Git Hub release ID.
+     */
+    readonly id: number;
```

Comment:
I would not know without the comment! 😂

Reply:
LOL. Yeah. I know it does get a little out of handle. Our stance is that if it is public (or exported in this case) then it is documented which TSLint is enforcing.

It's just easier to document it than have one off disable TSLint's everywhere.

---
In reply to: [416058245](https://github.com/microsoft/sarif-vscode-extension/pull/267#discussion_r416058245) [](ancestors = 416058245)

Reply:
I feel like rule "public == docs" came from a world with more than just 2 access modifiers.